CRM/Report/Form.php

```diff
@@ -4893,4 +4893,23 @@ public function joinGroupTempTable($baseTable, $field, $tableAlias) {
     }
   }
 
+  /**
+   * Get all labels for credit card type rows in report.
+   *
+   * @param string $creditCardTypes
+   *   comma separated option values of option group "credit_card_type".
+   */
+  public function getGroupCreditCardType($creditCardTypes) {
+    $cardTypes = CRM_Core_PseudoConstant::get('CRM_Financial_DAO_FinancialTrxn', 'card_type');
+    $types = explode(',', $creditCardTypes);
+    $labels = array();
+    foreach ($types as $value) {
+      $label = CRM_Utils_Array::value($value, $cardTypes);
+      if (!in_array($label, $labels)) {
+        $labels[] = $label;
+      }
+    }
+    return implode(', ', array_filter($labels));
+  }
+
```

Comment:
They normally have the labels applied to them during alterDisplay in individual reports. I couldn't find a report that does a group concat on an option value field, so I wrote this function to support it. I might have to change this function though, so that one can pass in an option group name to make it more generalised since I have a few other reports that are going to be needing this.